src/api/posts.ts

```diff
@@ -0,0 +1,53 @@
+import { BASE_URL } from './api';
+
+export const getAllPosts = ():Promise<Post[]> => {
+  return fetch(`${BASE_URL}/posts`)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Error');
+      }
+
+      return response.json();
+    });
+};
+
+export const getUserPosts = (userId: number):Promise<Post[]> => {
+  return getAllPosts()
+    .then(posts => {
+      return posts.filter((post: Post) => post.userId === userId);
+    });
+};
```

Comment:
use this ability of API for loading posts related to the chosen user 
![image](https://user-images.githubusercontent.com/64437769/133774462-06768979-5de1-42c5-b6a9-548ba8ab060a.png)